### DISCUSSION TITLE
Moving to SponsorWare

### DISCUSSION THREAD

**Discussion by facontidavide** [facontidavide/PlotJuggler#600] [Ideas]
With an increasing number of users using PlotJuggler, I decided to explore the option of [SponsorWare](https://github.com/sponsorware/docs).

Of course, I want to approach this in a way that doesn't hurt my current and future users, but gives a sufficient incentive to sponsor to support PlotJuggler.

A new private repository, called **PlotJuggler-SW** will be created and only sponsors in the [25$+ tier](https://github.com/sponsors/facontidavide) will have access to it.

The key idea is that all the new features will be released in the private repo and merged to the open source one only **3 months later**.

In more details...

## Open Repository

- Critical bugs are addressed immediately and merged to the Open Repository.
- Any external Pull Request is merged immediately.
- If a particular feature is sponsored by a company, it will be merged immediately.

## Private Repository

- Any new feature will be, eventually, merged to the Open Repository, 90 days later.

**Comment by MichaelOrlov** [comment]
@facontidavide Hi David.
Are you considering to change licensing model for **PlotJuggler**? 
Currently it's under the **GPL** license which is viral and prohibitive to use in (or with) commercial SW.
You have a great product which could be very valuable for many customers and developers.  
However, there are not to much benefits to get an early access to the fixes and features if product under the prohibitive licensing. 
It would be great to have it under the `Apache` or `MIT` license or have a dual licensing model.

**Comment by facontidavide** [comment]
@MichaelOrlov thank you for bringing this up.

First of all the license of PlotJuggler is LGPL, that is far less viral than GPL.
Furthermore, PlotJuggler uses Qt that is LPGL too, unless you purchase the commercial version.

> You have a great product which could be very valuable for many customers and developers.

I agree, and for this very same reason, I think that these customers should consider purchasing a license.

If anybody is interested to discuss this options, please reach me out :smile: 

And PLEASE, don't add "do it for free, think about the exposure". You correctly used the word "customers" :wink:

**Reply by MichaelOrlov** [reply]
@facontidavide As regards to the QT yes it's under the `LGPL` too. However it's under the dual licensing model and possible to by a commercial license.
Some times it's possible to use `QT` with `LGPL` license if using binaries without changes and loading them dynamically in runtime without static linking.  
But it's more difficult to workaround `LGPL` license for `PlotJuggler`.  
We might need to recompile it with our own data types and header files just to be able to use it as is. As well as to be able to fork it and make some modifications to use our proprietary SW IP blocks.
Are you open to discuss dual licensing model for `PlotJuggler`?

**Reply by facontidavide** [reply]
I am **very happy** to discuss dual licensing model, as long as one is LGPL and the other is "paid commercial license"

**Reply by MichaelOrlov** [reply]
@facontidavide Thank you for your response, we will take it in to consideration.
